src/editdist.jl

```diff
@@ -51,51 +58,174 @@ julia> edit_distance(g1, g2)
 ```
 """
 function edit_distance(G₁::AbstractGraph, G₂::AbstractGraph;
-                        insert_cost::Function=v -> 1.0,
-                        delete_cost::Function=u -> 1.0,
-                        subst_cost::Function=(u, v) -> 0.5,
-                        heuristic::Function=DefaultEditHeuristic)
+                        vertex_insert_cost=nothing,
+                        vertex_delete_cost=nothing,
+                        vertex_subst_cost=nothing,
+                        edge_insert_cost=nothing,
+                        edge_delete_cost=nothing,
+                        edge_subst_cost=nothing,
+                        heuristic=nothing)
+
+    if isnothing(vertex_insert_cost) &&
+            isnothing(vertex_delete_cost) &&
+            isnothing(vertex_subst_cost) &&
+            isnothing(edge_insert_cost) &&
+            isnothing(edge_delete_cost) &&
+            isnothing(edge_subst_cost) &&
+            isnothing(heuristic)
+        heuristic = default_edit_heuristic
+    end
+    if isnothing(vertex_insert_cost)
+        vertex_insert_cost = v -> 0.
+    end
+    if isnothing(vertex_delete_cost)
+        vertex_delete_cost = v -> 0.
+    end
+    if isnothing(vertex_subst_cost)
+        vertex_subst_cost = (u, v) -> 0.
+    end
+    if isnothing(edge_insert_cost)
+        edge_insert_cost = e -> 1.
+    end
+    if isnothing(edge_delete_cost)
+        edge_delete_cost = e -> 1.
+    end
+    if isnothing(edge_subst_cost)
+        edge_subst_cost = (e1, e2) -> 0.
+    end
+    if isnothing(heuristic)
+        heuristic = (λ, G₁, G₂) -> 0.
+    end
+    return _edit_distance(G₁::AbstractGraph, G₂::AbstractGraph,
+                        vertex_insert_cost,
+                        vertex_delete_cost,
+                        vertex_subst_cost,
+                        edge_insert_cost,
+                        edge_delete_cost,
+                        edge_subst_cost,
+                        heuristic)
+end
+
+function _edit_distance(G₁::AbstractGraph{T}, G₂::AbstractGraph{U},
+                        vertex_insert_cost::Function,
+                        vertex_delete_cost::Function,
+                        vertex_subst_cost::Function,
+                        edge_insert_cost::Function,
+                        edge_delete_cost::Function,
+                        edge_subst_cost::Function,
+                        heuristic::Function) where {T <: Integer, U <: Integer}
+
+    isdirected = is_directed(G₁) || is_directed(G₂)
 
-  # A* search heuristic
+    # compute the cost on edges due to associate u1 to v1 and u2 to v2
+    # u2 and v2 can eventually be 0
+    function association_cost(u1, u2, v1, v2)
+        cost = 0
```

Comment:
```suggestion
        cost = zero(Float64)
```
Might be slightly more type save than an integer.